### PR TITLE
Add ruby-install step for installing latest Ruby

### PR DESCRIPTION
**Pull request by jesperjohansson**
# Summary
Added Ruby installation step with ruby-install. Fixes latest Fastlane 2.137.0 not working due to system Ruby version (#65). 

## Test Plan

I have tested the Dockerfile by pushing to and using my [DockerHub repository](https://hub.docker.com/repository/docker/johanssonjesper/linux-react-native-android).

### What's required for testing (prerequisites)?
A CircleCI config (`.circleci/config.yml`).

### What are the steps to reproduce (after prerequisites)?
Add a custom executor and use it in your config instead of the react-native orb:
```language.yml
executors:
  custom:
    docker:
      - image: johanssonjesper/linux-react-native-android:latest
    environment:
      _JAVA_OPTIONS: "-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap"
      GRADLE_OPTS: '-Dorg.gradle.daemon=false -Dorg.gradle.jvmargs="-XX:+UnlockExperimentalVMOptions -XX:+HeapDumpOnOutOfMemoryError"'
      BUILD_THREADS: "2"
```

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ❌     |
| Android |    ✅     |

## Checklist

- [ ] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -32,8 +32,17 @@ RUN apt update -qq && apt install -qq -y --no-install-recommends \
         ruby-full \
         openssh-client \
         unzip \
+        libssl-dev \
```
Comment (gengjiawen) [inline Dockerfile]: I think it's better for this moved to the ruby install part.

File: Dockerfile [hunk 2]
```diff
@@ -32,8 +32,17 @@ RUN apt update -qq && apt install -qq -y --no-install-recommends \
...
+        libreadline-dev \
+        zlib1g-dev \
     && rm -rf /var/lib/apt/lists/*;
 
+# install ruby 2.4.9 for Fastlane usage
+RUN curl -L https://github.com/sstephenson/ruby-build/archive/v20191205.tar.gz | tar -zxvf - -C /tmp/ \
+  && cd /tmp/ruby-build-* && ./install.sh && cd / \
+  && ruby-build -v 2.4.9 /usr/local && rm -rfv /tmp/ruby-build-* \
```
Comment (gengjiawen) [inline Dockerfile]: I think just install the latest ruby version is okay.

Comment (jesperjohansson) [reply Dockerfile]: There is no "latest" argument for ruby-build that I can see but if you mean it should be 2.6.5 I can change it to that. I tried using ruby-install first but I wasn't able to replace system Ruby with it.

File: Dockerfile [hunk 3]
```diff
@@ -32,8 +32,17 @@ RUN apt update -qq && apt install -qq -y --no-install-recommends \
         ruby-full \
```
Comment (gengjiawen) [inline Dockerfile]: This line can be removed.

File: Dockerfile (hunk 2 above)
Comment (gengjiawen) [reply Dockerfile]: Maybe remove `ruby-full` from install and try ruby-install again ?

Comment (jesperjohansson) [reply Dockerfile]: Ah, it does seem to work now! I needed to add some dependencies. Does this look alright?
```
RUN apt-get update -qq \
  && apt-get install -qq -y --no-install-recommends \
          bison \
          zlib1g-dev \
          libyaml-dev \
          libssl-dev \
          libgdbm-dev \
          libreadline-dev \
          libncurses5-dev \
          libffi-dev \
  && curl -L https://github.com/postmodern/ruby-install/archive/v0.7.0.tar.gz | tar -zxvf - -C /tmp/ \
  && cd /tmp/ruby-install-* \
  && make install \
  && ruby-install --latest --system --cleanup ruby \
  && gem install bundler -N \
  && rm -rf /var/lib/apt/lists/*
```

Comment (jesperjohansson) [reply Dockerfile]: Should I make a new branch since this one has 2.4.9 in the name?

**Comment by gengjiawen** [comment]
thanks.

**Pull request merged**